task_2/list/linked_list.go

```diff
@@ -0,0 +1,126 @@
+package list
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+)
+
+//Node stores a value and a link to the next node in the list
+type Node struct {
+	value interface{} // Stores the value of the list
+	next  *Node       //Link to next node
+}
+
+// Implementing a linked list, stores a link to the first element of the list
+// and the current length
+type LinkedList struct {
+	head *Node //Link to the first element of the list
+	len  int   //current length of the list
+}
+
+//Adds an element at the beginning of the list
+//Takes a value of type interface{}
+func (l *LinkedList) Insert(value interface{}) {
+	link := l.head
+	l.head = &Node{value, link}
```

Comment:
(optional) you can add type checking.
in case user inserts `int` then `string` value, then we return error: incorrect type has been inserted